zfit/minimizers/fitresult.py

```diff
@@ -239,6 +242,50 @@ def _error(self, params, method, sigma):
                 raise KeyError("The following method is not a valid, implemented method: {}".format(method))
         return method(result=self, params=params, sigma=sigma)
 
+    def covariance(self, params: ParamsTypeOpt = None, as_dict: bool = False):
+        """Calculate the covariance matrix for `params`.
+
+            Args:
+                params (list(:py:class:`~zfit.Parameter`)): The parameters to calculate
+                    the covariance matrix. If `params` is `None`, use all *floating* parameters.
+                as_dict (bool): Default `False`. If `True` then returns a dictionnary.
+
+            Returns:
+                2D `numpy.array` of shape (N, N);
+                `dict`(param1, param2) -> covariance if `as_dict == True`.
+        """
+        params = self._input_check_params(params)
+        covariance_dict = self.minimizer._minuit_minimizer.covariance
+
+        cov = {}
+        for p1 in params:
+            for p2 in params:
+                key = (p1, p2)
+                cov[key] = covariance_dict[(k.name for k in key)]
+        covariance_dict = cov
+
+        if as_dict:
+            return covariance_dict
+        else:
+            return dict_to_matrix(params, covariance_dict)
+
+
+def dict_to_matrix(params, matrix_dict):
+
+    nparams = len(params)
+    matrix = np.empty((nparams, nparams))
+
+    for i in range(nparams):
+        pi = params[i]
+        for j in range(i, nparams):
+            pj = params[j]
+            key = (pi, pj)
```

Comment:
I've refactored things slightly to always use the `Parameter` objects and not the string names.

Reply:
Good, it makes sense to use the `Parameter`.